### PR TITLE
fix(kuma-cp) fix resource sync in K8S

### PR DESCRIPTION
**Pull request by lobkovilya**
### Summary

During preparing demo in Minikubes faced several issues. This PR aims to resolve them.

#### ACK/NACK didn't work as expected

In case of successful ACK we were putting
 ```go
s.latestACKed = s.latestReceived
``` 
which makes all types automatically ACKed (even if it is not). 
The same ACK method we have in XDS and MADS:
```go
func (s *Stream) ACK(typ string) error {
	latestReceived := s.latestReceived[typ]
	if latestReceived == nil {
		return nil
	}
	err := s.stream.Send(&envoy.DiscoveryRequest{
		VersionInfo:   latestReceived.VersionInfo,
		ResponseNonce: latestReceived.Nonce,
		ResourceNames: []string{},
		TypeUrl:       typ,
	})
	if err == nil {
		s.latestACKed = s.latestReceived
	}
	return err
}
```
Probably they also have to be fixed.

#### Bad filtering of Ingress on Remotes
I had to check Resource's zone with `localZone`. 

#### Restart of Global leads to situation when all Remotes hangs.

We spawned component on the first request from Global (OnStreamRequest). The spawned component was killed when stream is closed (OnStreamClosed). But when Global restarts on k8s new Pod starts when its replica is terminated and we have situation when OnStreamRequest from new Global comes earlier than OnStreamClosed. That leads to situation when spawned component is not active.

I get rid of stopping spawned component in OnStreamClosed. Remote needs Global for successful work, that's why stopping 
and restarting doesn't make sense.

### Documentation

Internal changes, documentation is not needed.

### CODE REVIEW

File: pkg/kds/store/sync.go [hunk 1]
```diff
@@ -75,6 +76,7 @@ func (s *syncResourceStore) Sync(upstream model.ResourceList, fs ...SyncOptionFu
 			downstream = filtered
 		}
 	}
+	s.log.Info("after filtering", "downstream", downstream)
```
Comment (nickolaev) [inline pkg/kds/store/sync.go:79]: How often are these messages appearing? We do not want to flood logs.

File: pkg/kds/remote/spawner.go [hunk 2]
```diff
@@ -30,22 +30,20 @@ type componentSpawner struct {
...
-		// spawn component
 		c.stop = make(chan struct{})
 		comp := c.factory(c.log.WithValues("streamID", streamID), req)
 		go func() {
 			err := comp.Start(c.stop)
-			c.log.Error(err, "component finished with an error")
-			atomic.CompareAndSwapInt32(&c.spawned, 1, 0)
+			if err != nil {
+				c.log.Error(err, "component finished with an error")
+			}
```
Comment (nickolaev) [inline pkg/kds/remote/spawner.go:39]: That is better, indeed,

File: pkg/kds/remote/components.go [hunk 3]
```diff
@@ -81,15 +81,15 @@ func providedFilter(clusterName string) reconcile.ResourceFilter {
...
+func Callbacks(syncer sync_store.ResourceSyncer, k8sStore bool, localZone string) *kds_client.Callbacks {
 	return &kds_client.Callbacks{
 		OnResourcesReceived: func(clusterID string, rs model.ResourceList) error {
 			if k8sStore && rs.GetItemType() != mesh.MeshType && rs.GetItemType() != system.SecretType {
 				util.AddSuffixToNames(rs.GetItems(), "default")
 			}
 			if rs.GetItemType() == mesh.DataplaneType {
 				return syncer.Sync(rs, sync_store.PrefilterBy(func(r model.Resource) bool {
-					return r.(*mesh.DataplaneResource).Spec.IsIngress() && util.ZoneTag(r) == clusterID
+					return r.(*mesh.DataplaneResource).Spec.IsIngress() && localZone != util.ZoneTag(r)
```
Comment (nickolaev) [inline pkg/kds/remote/components.go:92]: 👍

File: pkg/kds/client/stream.go [hunk 4]
```diff
@@ -79,7 +79,7 @@ func (s *stream) ACK(typ string) error {
 		TypeUrl: typ,
 	})
 	if err == nil {
-		s.latestACKed = s.latestReceived
+		s.latestACKed[typ] = latestReceived
```
Comment (nickolaev) [inline pkg/kds/client/stream.go:82]: OK, that is simple.

File: pkg/kds/client/sink.go [hunk 5]
```diff
@@ -78,10 +78,10 @@ func (s *kdsSink) Start(stop <-chan struct{}) (errs error) {
 		if err != nil {
 			return errors.Wrap(err, "failed to receive a discovery response")
 		}
-		s.log.Info("DiscoveryResponse received", "type", rs.GetItemType())
+		s.log.Info("DiscoveryResponse received", "response", rs)
```
Comment (jakubdyszkiewicz) [inline pkg/kds/client/sink.go:81]: Does it print full response? We are syncing secrets here. I don't want to log secret here

**Pull request merged**